library/pk_wrap.c

```diff
@@ -480,26 +482,25 @@ static int rsa_alt_decrypt_wrap( void *ctx,
 }
 
 #if defined(MBEDTLS_RSA_C)
-static int rsa_alt_check_pair( const mbedtls_pk_context *pub, const void *prv )
+static int rsa_alt_check_pair( const mbedtls_pk_context *pub,
+                               const mbedtls_pk_context *prv )
 {
     unsigned char sig[MBEDTLS_MPI_MAX_SIZE];
     unsigned char hash[32];
     size_t sig_len = 0;
     int ret;
-    const mbedtls_pk_context* prv_context = prv;
-
-    if( prv_context->pk_info->type == MBEDTLS_PK_RSA_ALT )
+    if( prv->pk_info->type == MBEDTLS_PK_RSA_ALT )
```

Comment:
Again, this line is redundant, the condition is always true when this function is called. Other `xxx_check_pair()` function directly cast `prv->pk_ctx` to the respective type without checking, I don't see why we should check here. Please remove this line and only keep what's inside the "then" block.

Reply:
Done